src/carousel/carousel.ts

```diff
@@ -111,13 +111,25 @@ export class NgbSlide {
 				<ng-template [ngTemplateOutlet]="slide.tplRef"></ng-template>
 			</div>
 		</div>
-		<button class="carousel-control-prev" type="button" (click)="arrowLeft()" *ngIf="showNavigationArrows">
+		<button
+			class="carousel-control-prev"
+			type="button"
+			(click)="arrowLeft()"
+			*ngIf="showNavigationArrows"
+			aria-labelledby="button-span-previous"
+		>
 			<span class="carousel-control-prev-icon" aria-hidden="true"></span>
-			<span class="visually-hidden" i18n="@@ngb.carousel.previous">Previous</span>
+			<span class="visually-hidden" i18n="@@ngb.carousel.previous" id="button-span-previous">Previous</span>
 		</button>
-		<button class="carousel-control-next" type="button" (click)="arrowRight()" *ngIf="showNavigationArrows">
+		<button
+			class="carousel-control-next"
+			type="button"
+			(click)="arrowRight()"
+			*ngIf="showNavigationArrows"
+			aria-labelledby="button-span-next"
+		>
 			<span class="carousel-control-next-icon" aria-hidden="true"></span>
-			<span class="visually-hidden" i18n="@@ngb.carousel.next">Next</span>
+			<span class="visually-hidden" i18n="@@ngb.carousel.next" id="button-span-next">Next</span>
```

Comment:
This will not be unique in case there are several carousels on the page

Reply:
indeed, I am gonna use the same logic as in accordion for the generated id

Reply:
I made the change to add unique ids for the spans. Should we use that ids (that are generated for each carousel) also as id on the carousel itself as we do for the accordion with the input?